django_mongodb_backend/cache.py

```diff
@@ -84,7 +92,13 @@ def get_many(self, keys, version=None):
         with self.collection_for_read.find(
             {"key": {"$in": tuple(keys_map)}, **self._filter_expired(expired=False)}
         ) as cursor:
-            return {keys_map[row["key"]]: self.serializer.loads(row["value"]) for row in cursor}
+            results = {}
+            for row in cursor:
+                try:
+                    results[keys_map[row["key"]]] = self.serializer.loads(row["value"])
+                except (BadSignature, TypeError):
+                    self.delete(row["key"])
```

Comment:
I'm having trouble understanding why we're catching `BadSignature` and `TypeError` issues. What would trigger this and why would it be okay to delete a row in these instances?